instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py

```diff
@@ -0,0 +1,653 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Instrument to report system (CPU, memory, network) and
+process (CPU, memory, garbage collection) metrics. By default, the
+following metrics are configured:
+
+.. code:: python
+
+    {
+        "system.cpu.time": ["idle", "user", "system", "irq"],
+        "system.cpu.utilization": ["idle", "user", "system", "irq"],
+        "system.memory.usage": ["used", "free", "cached"],
+        "system.memory.utilization": ["used", "free", "cached"],
+        "system.swap.usage": ["used", "free"],
+        "system.swap.utilization": ["used", "free"],
+        "system.disk.io": ["read", "write"],
+        "system.disk.operations": ["read", "write"],
+        "system.disk.time": ["read", "write"],
+        "system.network.dropped.packets": ["transmit", "receive"],
+        "system.network.packets": ["transmit", "receive"],
+        "system.network.errors": ["transmit", "receive"],
+        "system.network.io": ["trasmit", "receive"],
+        "system.network.connections": ["family", "type"],
+        "runtime.memory": ["rss", "vms"],
+        "runtime.cpu.time": ["user", "system"],
+    }
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry._metrics import set_meter_provider
+    from opentelemetry.instrumentation.system_metrics import SystemMetricsInstrumentor
+    from opentelemetry.sdk._metrics import MeterProvider
+    from opentelemetry.sdk._metrics.export import ConsoleMetricExporter, PeriodicExportingMetricReader
+
+    exporter = ConsoleMetricExporter()
+
+    set_meter_provider(MeterProvider(PeriodicExportingMetricReader(exporter)))
+    SystemMetricsInstrumentor().instrument()
+
+    # metrics are collected asynchronously
+    input("...")
+
+    # to configure custom metrics
+    configuration = {
+        "system.memory.usage": ["used", "free", "cached"],
+        "system.cpu.time": ["idle", "user", "system", "irq"],
+        "system.network.io": ["trasmit", "receive"],
+        "runtime.memory": ["rss", "vms"],
+        "runtime.cpu.time": ["user", "system"],
+    }
+    SystemMetricsInstrumentor(config=configuration).instrument()
+
+API
+---
+"""
+
+import gc
+import os
+from platform import python_implementation
+from typing import Collection, Dict, Iterable, List, Optional
+
+import psutil
+
+from opentelemetry._metrics import get_meter
+from opentelemetry._metrics.measurement import Measurement
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.system_metrics.package import _instruments
+from opentelemetry.instrumentation.system_metrics.version import __version__
+from opentelemetry.sdk.util import get_dict_as_key
+
+
+class SystemMetricsInstrumentor(BaseInstrumentor):
+    # pylint: disable=too-many-statements
+    def __init__(
+        self,
+        labels: Optional[Dict[str, str]] = None,
```

Comment:
It looks like this just adds extra attributes to all measurements. We should probably remove this?

We can do it in a separate PR though as this is just restoring the package

Reply:
Agree, happy to do this as a separate item

Reply:
Added an issue: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1034